colab_train.ipynb

```diff
@@ -1 +1,231 @@
ta":{"colab":{"base_uri":"https://localhost:8080/"},"id":"d514HD0J2xmy","executionInfo":{"status":"ok","timestamp":1615509979883,"user_tz":480,"elapsed":1619,"user":{"displayName":"Robert Lee","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14Gg5Bo21GMlIXvvTRzHk7PUzZL8VXUJiObN8raRq=s64","userId":"05735836358878398466"}},"outputId":"2a6a0676-17de-44c1-f513-69b66f191cf4"},"source":["!ls /content/datasets/coco/train2017 | wc -l\n","!ls /content/datasets/coco/val2017 | wc -l"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"SBQbXIcUC4sG","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1615512633807,"user_tz":480,"elapsed":2389,"user":{"displayName":"Robert Lee","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14Gg5Bo21GMlIXvvTRzHk7PUzZL8VXUJiObN8raRq=s64","userId":"05735836358878398466"}},"outputId":"17f9fa33-cd45-4f7c-a328-df6b9ae24f00"},"source":["# Ensure that filesystem is set up correctly and will not be the bottleneck\n","import time\n","from datetime import datetime, timedelta\n","\n","start = time.time()\n","\n","import os\n","def print_files_in_dir(path):\n","    print(len(os.listdir(path)), \" files at \", path)\n","\n","print_files_in_dir('/content/datasets/coco/train2017/')\n","print_files_in_dir('/content/datasets/coco/val2017/')\n","\n","\n","setup_time = time.time() - start\n","print(\"Total setup time: {}\".format(str(timedelta(seconds=setup_time))))\n","\n","if setup_time > 1:\n","    print(\"There appears to be a bottleneck with filesystem loading time. This may severely impact training speed.\")"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"UeLJoy2iz5sE","executionInfo":{"status":"ok","timestamp":1615508555081,"user_tz":480,"elapsed":5955,"user":{"displayName":"Robert Lee","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14Gg5Bo21GMlIXvvTRzHk7PUzZL8VXUJiObN8raRq=s64","userId":"05735836358878398466"}},"outputId":"5fb5bd9a-d2f3-4e0a-d5b2-d18acdccd85b"},"source":["# Change dir to project dir\n","%cd \"/content/gdrive/MyDrive/Colab Data/COCO-Human-Pose\"\n","\n","# Toggle COLAB_TRAINING variable in constants file\n","# https://askubuntu.com/questions/20414/find-and-replace-text-within-a-file-using-commands\n","!sed -i.bak 's/COLAB_TRAINING = False/COLAB_TRAINING = True/g' constants.py \n","\n","# Remove backup file\n","!rm constants.py.bak"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"O2MRZcYU2TN2","executionInfo":{"status":"ok","timestamp":1615512475320,"user_tz":480,"elapsed":668715,"user":{"displayName":"Robert Lee","photoUrl":"https://lh3.googleusercontent.com/a-/AOh14Gg5Bo21GMlIXvvTRzHk7PUzZL8VXUJiObN8raRq=s64","userId":"05735836358878398466"}},"outputId":"1f977904-3e1f-4e7d-c0ac-3d8511514c14"},"source":["!python3 train.py --epochs 100 --batch 12 --hourglass 4"],"execution_count":null,"outputs":[]}]}
```

Comment:
<p>Thx for this comment</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/robertklee/COCO-Human-Pose/pull/8/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>